worlds/sc2/options.py

```diff
@@ -1093,6 +1092,17 @@ def get_excluded_missions(world: 'SC2World') -> Set[SC2Mission]:
     # Omitting missions not in enabled campaigns
     for campaign in disabled_campaigns:
         excluded_missions = excluded_missions.union(campaign_mission_table[campaign])
+    # Omitting unwanted mission variants
+    if get_option_value(world, "enable_race_swap") == EnableRaceSwapVariants.option_pick_one:
+        swaps: Set[SC2Mission] = set([mission for mission in set(SC2Mission).difference(excluded_missions) if mission.flags & MissionFlag.HasRaceSwap|MissionFlag.RaceSwap])
```

Comment:
I think this may cause an issue where generating with the same seed can produce different results. You probably need to go through the world's / multiworld's random number generator.

Reply:
Hm... didn't know that would be an issue. I'll look into it

Reply:
Refactored so random removal only occurs via world.random